components/server/src/code-sync/code-sync-service.ts

```diff
@@ -253,24 +253,10 @@ export class CodeSyncService {
                 if (response.status !== 200) {
                     throw new Error(`code sync: blob service: upload failed with ${response.status} ${response.statusText}`);
                 }
-                oldObject = oldRev && toObjectName(resourceKey, oldRev);
+                oldRevList = oldRevs
             }, { latestRev, revLimit });
-            if (oldObject) {
-                const request = new DeleteRequest();
-                request.setOwnerId(userId);
-                request.setExact(oldObject);
-
-                const blobsClient = this.blobsProvider.getDefault();
-                blobsClient.delete(request, (err: ServiceError | null) => {
-                    if (err) {
-                        if (err.code === status.NOT_FOUND) {
-                            // we're good here
-                            return;
-                        }
-                        log.error({ userId }, 'code sync: failed to delete', err, { object: oldObject });
-                    }
-                });
-            }
+            // sync delete old revs from storage
+            this.deleteObjects(userId, resourceKey, oldRevList)
```

Comment:
await?

Reply:
We don't need to await this delete func, unless client will fetch list immediately after post their configuration